lib/dockerregistry/inventory.go

```diff
@@ -1033,6 +1022,22 @@ func (sc *SyncContext) ReadRegistries(
 	sc.ExecRequests(populateRequests, processRequest)
 }
 
+func SplitByKnownRegistries(
+	r RegistryName,
+	rcs []RegistryContext) (RegistryName, ImageName, error) {
+
+	for _, rc := range rcs {
+		//		if inameIdx := strings.LastIndex(string(r), string(rc.Name)); inameIdx != -1 {
+		//			return rc.Name, ImageName(r[inameIdx+1:]), nil
+		//		}
+		if trimmed := strings.TrimPrefix(string(r), string(rc.Name)); len(trimmed) < len(r) {
+			return rc.Name, ImageName(trimmed[1:]), nil
```

Comment:
Is this `[1:]` because we expected it not to have a slash?  If so I think that needs a comment and probably it's safer not to assume that rc.Name is formatted without a slash, and to check that the suffix does start with a slash, etc...

Reply:
Yes, exactly. ACK